### PR TITLE
add advertising state to address

### PR DESCRIPTION
**Pull request by patrickweh**

### CODE REVIEW

File: database/migrations/2024_12_17_121217_add_advertising_state_to_addresses_table.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration
+{
+    public function up(): void
+    {
+        Schema::table('addresses', function (Blueprint $table) {
+            $table->string('advertising_state')->default('open')->after('contact_id');
```
Comment (SirSplasch) [inline database/migrations/2024_12_17_121217_add_advertising_state_to_addresses_table.php:12]: after 'password'

File: src/Models/Address.php [hunk 2]
```diff
@@ -201,6 +204,7 @@ protected static function booted(): void
     protected function casts(): array
     {
         return [
+            'advertising_state' => AdvertisingState::class,
```
Comment (SirSplasch) [inline src/Models/Address.php:207]: after date_of_birth

File: src/Rulesets/Address/CreateAddressRuleset.php [hunk 3]
```diff
@@ -39,6 +41,10 @@ public function rules(): array
                 'nullable',
                 app(ModelExists::class, ['model' => Language::class]),
             ],
+            'advertising_state' => [
+                'string',
```
Comment (SirSplasch) [inline src/Rulesets/Address/CreateAddressRuleset.php:45]: nullable?

File: src/Rulesets/Address/UpdateAddressRuleset.php [hunk 4]
```diff
@@ -39,6 +41,10 @@ public function rules(): array
                 'nullable',
                 app(ModelExists::class, ['model' => Language::class]),
             ],
+            'advertising_state' => [
+                'string',
```
Comment (SirSplasch) [inline src/Rulesets/Address/UpdateAddressRuleset.php:45]: nullable?

File: src/States/Order/OrderState.php [hunk 5]
```diff
@@ -85,9 +84,4 @@ public static function getFrontendFormatter(...$args): string|array
                 ->map(fn ($key) => (new $key(''))->color()),
         ];
     }
-
-    public function toArray(): array|string
```
Comment (SirSplasch) [inline src/States/Order/OrderState.php:89]: does removing this have any effect on the existing code-base. Or was this just added but never used?

File: src/Livewire/Contact/Addresses.php [hunk 6]
```diff
@@ -40,8 +40,20 @@ class Addresses extends Component
...
 
+    public array $availableStates = [];
+
     public function mount(): void
     {
+        $this->availableStates = app(Address::class)
+            ->getStatesFor('advertising_state')
+            ->map(function (string $state) {
+                return [
+                    'label' => __(ucfirst(str_replace('_', ' ', $state))),
```
Comment (SirSplasch) [inline src/Livewire/Contact/Addresses.php:51]: why not __(Str::headline($state))?

**Review by SirSplasch — CHANGES_REQUESTED** [review]

File: src/Rulesets/Address/CreateAddressRuleset.php (hunk 3 above)
Comment (patrickweh) [reply src/Rulesets/Address/CreateAddressRuleset.php:45]: would keep it not nullable as the column isnt nullable

File: src/States/Order/OrderState.php (hunk 5 above)
Comment (patrickweh) [reply src/States/Order/OrderState.php:89]: the to array is in the flux state base class, as well as the arrayable contract

**Review by SirSplasch — APPROVED** [review]

**Pull request merged**